### PR TITLE
A round of data libraries improvements.

### PR DESCRIPTION
**Pull request by martenson**
* *add new PATCH API for modifying library datasets which operates on the connected (latest) ldda*
* add frontend functionality to leverage new API
* use more of base manager
* renamed `lda_datasets.py` to `library_datasets.py` to remove confusion
* remove unused code

### CODE REVIEW

**Comment by martenson** [comment]
once tests finish this should be ready for review

**Comment by bgruening** [comment]
This is super cool Martin!

**Comment by jmchilton** [comment]
Started working on tests for libraries to test this against (https://github.com/galaxyproject/galaxy/pull/4543) but I ran out of steam and got stumped on a couple odd issues. In manual testing everything seemed fine though. Thanks for the work @martenson!

File: lib/galaxy/managers/library_datasets.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        """
+        Update the given library dataset - the latest linked ldda.
+        Updating older lddas (versions) is not supported.
+
+        :param  ld: library dataset to change
+        :type   ld: LibraryDataset
+        :param  check_ownership:           flag whether to check that user owns the item
+        :type   check_ownership:           bool
+        :param  check_accessible:          flag whether to check that user can access item
+        :type   check_accessible:          bool
```
Comment (nsoranzo) [inline lib/galaxy/managers/library_datasets.py:53]: `check_ownership` and `check_accessible` are not parameters of this function, documentation for `payload` is missing instead.

File: lib/galaxy/managers/library_datasets.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+        :raises:    ObjectNotFound
+        """
+        if trans.user_is_admin():
+            # all operations are available to an admin
+            return ld
+        if not trans.app.security_agent.can_modify_library_item(trans.get_current_user_roles(), ld):
+            raise InsufficientPermissionsException('You do not have proper permission to modify this library dataset.')
+        elif ld.deleted:
+            raise ObjectNotFound('Library dataset with the id provided is deleted.')
```
Comment (nsoranzo) [inline lib/galaxy/managers/library_datasets.py:178]: I think the check on `deleted` should be the first, admins are not allowed to modify a deleted library dataset anyway by the check on line 62. So I think this should be:
```python
    if ld.deleted:
        raise ...
    if trans.user_is_admin():
        return ld
    if not trans.app.security_agent...
        raise ...
```

File: lib/galaxy/managers/library_datasets.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+        :returns:   the changed library dataset
+        :rtype:     galaxy.model.LibraryDataset
+
+        :raises:    ItemAccessibilityException, InsufficientPermissionsException
+        """
+        current_user_roles = trans.get_current_user_roles()
+        self.check_modifiable(trans, ld)
+        if ld.deleted is True:
+            raise ItemAccessibilityException("You cannot update a deleted library dataset. Undelete it first.")
```
Comment (nsoranzo) [inline lib/galaxy/managers/library_datasets.py:63]: If `check_modifiable` is modified as suggested below, this check becomes redundant.

**Review by nsoranzo — COMMENTED** [review]
Sorry for the late review, too many incoming PRs lately (which is obviously good!).

**Pull request merged**